src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -753,6 +753,52 @@ public static void M1(in int arg1, in int arg2, in int arg3)
         }
 
         [Fact]
+        public void ReadonlyParamAsyncSpillIn()
+        {
+            var text = @"
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Test().Wait();
+        }
+
+        public static async Task Test()
+        {
+            int local = 1;
+            M1(in RefReturning(ref local), await GetT(2), 3);
+        }
+
+        private static ref int RefReturning(ref int arg)
+        {
+            return ref arg;
+        }    
+
+        public static async Task<T> GetT<T>(T val)
+        {
+            await Task.Yield();
+            return val;
+        }
+
+        public static void M1(in int arg1, in int arg2, in int arg3)
+        {
+            System.Console.WriteLine(arg1 + arg2 + arg3);
+        }
+    }
+
+";
+
+            var comp = CreateCompilationWithMscorlib46(text, new[] { ValueTupleRef, SystemRuntimeFacadeRef }, options: TestOptions.ReleaseExe);
+            comp.VerifyEmitDiagnostics(
+                // (14,44): error CS8178: 'await' cannot be used in an expression containing a call to 'Program.RefReturning(ref int)' because it returns by reference
```

Comment:
So this works if you don't specify `in` because we can make a copy over the await boundary? If so... this is kind of a bad error message. It's a corner case, but really confusing.

Reply:
In reality this is a very rare error. Yes, we can improve the error message. I'd rather do it in a separate change though.

---
In reply to: [141985353](https://github.com/dotnet/roslyn/pull/22405#discussion_r141985353) [](ancestors = 141985353)

Reply:
[https://github.com/dotnet/roslyn/issues/22445](https://github.com/dotnet/roslyn/issues/22445)

---
In reply to: [141985636](https://github.com/dotnet/roslyn/pull/22405#discussion_r141985636) [](ancestors = 141985636,141985353)